### PR TITLE
Code Coverage for Linkage

### PR DESCRIPTION
**Pull request by KennethSkylight**
[ClickUp Ticket](https://reportstream-service-request.clickup.com/t/3g9pbgz)

To increase the code coverage for linkage unit tests. The `overwrite = False` needed to be tested. This PR adds two tests for both bundle and resource linkage as well as fixing a logic bug on the linkage bundle

### CODE REVIEW

File: tests/fhir/linkage/test_fhir_link.py [hunk 1]
```diff
@@ -85,3 +85,67 @@ def test_patient_identifier():
             assert len(resource["identifier"]) == 2
             assert resource["identifier"][-1] == expected_new_identifier
             break
+
+
+def test_bundle_overwrite():
```
Comment (spencer-skylight) [inline tests/fhir/linkage/test_fhir_link.py:90]: Nit: can you make the name more descriptive?  Maybe copy the name of the non-overwrite and append _overwrite? I know we can glean some context from the file, but the link module could do other things with bundles that we might want to differentiate in the future.

File: tests/fhir/linkage/test_fhir_link.py [hunk 2]
```diff
@@ -85,3 +85,67 @@ def test_patient_identifier():
...
+        + "1234 Silversun Strip Zakera Ward, Citadel 99999"
+    )
+
+    expected_new_identifier = {
+        "value": generate_hash_str(plaintext, salt_str),
+        "system": "urn:ietf:rfc:3986",
+        "use": "temp",
+    }
+
+    new_bundle = add_patient_identifier_in_bundle(incoming_bundle, salt_str, False)
```
Comment (spencer-skylight) [inline tests/fhir/linkage/test_fhir_link.py:112]: I think you've got the `overwrite` - non-`overwrite` test cases inverted.  I think if you omit the last parameter here, and change the other test cases' calls to `False`, you should be good.

File: tests/fhir/linkage/test_fhir_link.py [hunk 3]
```diff
@@ -85,3 +85,67 @@ def test_patient_identifier():
...
+    assert new_bundle != incoming_bundle
+    assert len(new_bundle["entry"]) == 3
+
+    for resource in new_bundle["entry"]:
+        if resource["resource"]["resourceType"] == "Patient":
+            assert len(resource["resource"]["identifier"]) == 2
+            assert resource["resource"]["identifier"][-1] == expected_new_identifier
+
+
+def test_overwrite():
```
Comment (spencer-skylight) [inline tests/fhir/linkage/test_fhir_link.py:122]: Same nit as previous function.

File: tests/fhir/linkage/test_fhir_link.py [hunk 4]
```diff
@@ -85,3 +85,67 @@ def test_patient_identifier():
...
+    expected_new_identifier = {
+        "value": generate_hash_str(plaintext, salt_str),
+        "system": "urn:ietf:rfc:3986",
+        "use": "temp",
+    }
+
+    for entry in incoming_bundle["entry"]:
+        if entry["resource"]["resourceType"] == "Patient":
+            resource = entry.get("resource")
+            new_resource = add_patient_identifier(resource, salt_str, False)
```
Comment (spencer-skylight) [inline tests/fhir/linkage/test_fhir_link.py:147]: Same comment as previous function.

**Review by spencer-skylight — CHANGES_REQUESTED** [review]
Looking good, just a few minor changes to suggest.

**Review by spencer-skylight — APPROVED** [review]
LGTM!

**Review by bamader — APPROVED** [review]
LGTM!

**Pull request merged**